core/scripts/sql/texera_ddl.sql

```diff
@@ -1,286 +1,351 @@
+-- 2. Connect to the new database (psql only)
+-- ============================================
+\c texera_db
+
+CREATE SCHEMA IF NOT EXISTS texera_db;
+SET search_path TO texera_db, public;
+
+-- ============================================
+-- 3. Drop all tables if they exist
+--    (CASCADE handles FK dependencies)
+-- ============================================
+DROP TABLE IF EXISTS operator_executions CASCADE;
+DROP TABLE IF EXISTS operator_port_executions CASCADE;
+DROP TABLE IF EXISTS workflow_user_access CASCADE;
+DROP TABLE IF EXISTS workflow_of_user CASCADE;
+DROP TABLE IF EXISTS user_config CASCADE;
+DROP TABLE IF EXISTS "user" CASCADE;
+DROP TABLE IF EXISTS workflow CASCADE;
+DROP TABLE IF EXISTS workflow_version CASCADE;
+DROP TABLE IF EXISTS project CASCADE;
+DROP TABLE IF EXISTS workflow_of_project CASCADE;
+DROP TABLE IF EXISTS workflow_executions CASCADE;
+DROP TABLE IF EXISTS dataset CASCADE;
+DROP TABLE IF EXISTS dataset_user_access CASCADE;
+DROP TABLE IF EXISTS dataset_version CASCADE;
+DROP TABLE IF EXISTS public_project CASCADE;
+DROP TABLE IF EXISTS project_user_access CASCADE;
+DROP TABLE IF EXISTS workflow_user_likes CASCADE;
+DROP TABLE IF EXISTS workflow_user_clones CASCADE;
+DROP TABLE IF EXISTS workflow_view_count CASCADE;
+DROP TABLE IF EXISTS workflow_user_activity CASCADE;
+DROP TABLE IF EXISTS user_activity CASCADE;
+DROP TABLE IF EXISTS dataset_user_likes CASCADE;
+DROP TABLE IF EXISTS dataset_view_count CASCADE;
+
+-- ============================================
+-- 4. Create PostgreSQL enum types
+--    to mimic MySQL ENUM fields
+-- ============================================
+DROP TYPE IF EXISTS user_role_enum CASCADE;
+DROP TYPE IF EXISTS privilege_enum CASCADE;
+
+CREATE TYPE user_role_enum AS ENUM ('INACTIVE', 'RESTRICTED', 'REGULAR', 'ADMIN');
+CREATE TYPE privilege_enum AS ENUM ('NONE', 'READ', 'WRITE');
+
+-- ============================================
+-- 5. Create tables
+-- ============================================
+
+-- "user" table
+CREATE TABLE IF NOT EXISTS "user"
 (
-    `uid`           INT UNSIGNED AUTO_INCREMENT NOT NULL,
-    `name`          VARCHAR(256)                NOT NULL,
-    `email`         VARCHAR(256) UNIQUE,
-    `password`      VARCHAR(256),
-    `google_id`     VARCHAR(256) UNIQUE,
-    `role`          ENUM('INACTIVE', 'RESTRICTED', 'REGULAR', 'ADMIN') NOT NULL DEFAULT 'INACTIVE',
-    `google_avatar` VARCHAR(100) null,
-    PRIMARY KEY (`uid`),
-    CONSTRAINT CK_nulltest
-        CHECK (`password` IS NOT NULL OR `google_id` IS NOT NULL)
-) ENGINE = INNODB,
--- start auto increment userID from 1 because userID 0 means user not exists
-  AUTO_INCREMENT = 1;
-
+    uid           SERIAL PRIMARY KEY,
+    name          VARCHAR(256) NOT NULL,
+    email         VARCHAR(256) UNIQUE,
+    password      VARCHAR(256),
+    google_id     VARCHAR(256) UNIQUE,
+    role          user_role_enum NOT NULL DEFAULT 'INACTIVE',
+    google_avatar VARCHAR(100),
```

Comment:
Can you swap `role` and `google_avatar` for me with this chance?

Reply:
swapped